tools/mochaRunner/src/utils.ts

```diff
@@ -95,18 +95,17 @@ export function findEffectiveExpectationForTest(
   expectations: TestExpectation[],
   result: MochaTestResult
 ): TestExpectation | undefined {
-  return expectations
-    .filter(expectation => {
-      return (
-        '' === expectation.testIdPattern ||
-        getTestId(result.file) === expectation.testIdPattern ||
-        getTestId(result.file, result.fullTitle) === expectation.testIdPattern
-      );
-    })
-    .pop();
+  return expectations.find(expectation => {
+    return (
+      '' === expectation.testIdPattern ||
```

Comment:
do you want to require `*` and `[spec]*` instead of the current notation? It think it would be easier to maintain if we only support explicit *